### PR TITLE
Preserve the tasks & configuration while adding .net specific ones

### LINKED ISSUE #1993: Banner overrides .vscode settings if just added by other banner
## Environment data
`dotnet --info` output:
```
**.NET Command Line Tools (2.0.0)

Product Information:
 Version:            2.0.0
 Commit SHA-1 hash:  cdcd1928c9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x86
 Base Path:   C:\Program Files (x86)\dotnet\sdk\2.0.0\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d
```

VS Code version: `1.19.3`
C# Extension version: `1.13.1`

## Steps to reproduce
1. We are building an extension for [Connected Environment for AKS](https://review.docs.microsoft.com/en-us/visualstudio/connected-environment/visual-studio-connected-environment?branch=johnsta)
2. Setup an environment as described here for [get-started-netcore](https://review.docs.microsoft.com/en-us/visualstudio/connected-environment/get-started-netcore-01?branch=johnsta)
3. Once you see the banner like below:
image
4. Click on `Yes` for `[Connected Environment]` first to generate `tasks.json` + `launch.json` inside `.vscode`
5. Then click on `Yes` other banner.

## Expected  behavior
Banner should update existing `tasks.json` + `launch.json` with the C# debugger settings and it should contain both the settings of C# & Connected Environment. 

## Actual behavior
Banner overrides the  `tasks.json` + `launch.json` files content created by our banner.

### PR DESCRIPTION
**Pull request by vishrutshah**
Fixes #1993

### CODE REVIEW

File: src/assets.ts [hunk 1]
```diff
@@ -419,9 +419,16 @@ export function addTasksJsonIfNecessary(generator: AssetGenerator, operations: O
             return resolve();
         }
 
+        // Read existing Tasks configuration
+        const tasksConfigs = vscode.workspace.getConfiguration('tasks');
+        let existingTaskConfigs = tasksConfigs.get<Array<tasks.TaskDescription>>('tasks');
```
[the remark this answers is not in the document]
Comment (vishrutshah) [reply src/assets.ts]: I've verified with having no `task.json` at all but let me also verify if `tasks.json` is having empty content :-)

[the remark this answers is not in the document]
Comment (vishrutshah) [reply src/assets.ts]: Ok. So after testing both the scenarios, looks like we do not need addition if statements around 
```
let existingTaskConfigs = tasksConfigs.get<Array<tasks.TaskDescription>>('tasks');
```

Case 1: When we do not have `tasks.json` then `getConfiguration` returns an object on which we call `get<T>` which returns `undefined`
Case 2: When we have `tasks.json` with empty array of tasks then we do not visit this path as we existing logic `resolve({ updateTasksJson: buildTasks.length === 0 });`  would prevent from modifying the tasks.json

Let me know your thoughts?

**Comment by vishrutshah** [comment]
@gregg-miskelly Could you please help me see what I may doing wrong here as CI is failing? Thanks!

**Comment by vishrutshah** [comment]
Aha seems like I was missing updates from upstream :-)

File: src/assets.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gregg-miskelly) [reply src/assets.ts]: @vishrutshah your analysis sounds correct to me.

[the remark this answers is not in the document]
Comment (vishrutshah) [reply src/assets.ts]: Cool. Let me know if any more changes are required. Happy to do it :)

File: src/assets.ts [hunk 2]
```diff
@@ -442,11 +449,27 @@ function addLaunchJsonIfNecessary(generator: AssetGenerator, operations: Operati
...
+
         const isWebProject = generator.hasWebServerDependency();
-        const launchJson: string = generator.createLaunchJson(isWebProject);
+        let launchJson: string = generator.createLaunchJson(isWebProject);
+
+        if (existingLaunchConfigs) {
+            let existingLaunchConfigsString = JSON.stringify(existingLaunchConfigs, null, '    ');
+            const lastBracket = launchJson.lastIndexOf(']');
+            const lastBracketInExistingConfig = existingLaunchConfigsString.lastIndexOf(']');
+            const firstBracketInExistingConfig = existingLaunchConfigsString.lastIndexOf('[');
```
Comment (gregg-miskelly) [inline src/assets.ts]: On this line, don't you want `existingLaunchConfigsString.indexOf('[');` instead of `lastIndexOf`?

**Comment by gregg-miskelly** [comment]
Otherwise LGTM. @DustinCampbell @TheRealPiotrP would either of you like to review as well?

File: src/assets.ts (hunk 2 above)
Comment (vishrutshah) [reply src/assets.ts]: Indeed. My mistake. It did work as there was only one occurrence of `[`. Thanks for catching. Let me fix this

**Comment by vishrutshah** [comment]
Thanks @gregg-miskelly for helping review the changes!

**Pull request merged**